core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -62,6 +68,29 @@ oppia.directive('supplementalCard', [
             $scope.helpCardHasContinueButton = false;
           };
 
+          $scope.showHint = function() {
+            var isSupplementalCard = true;
+            $scope.helpCardHtml = HintManagerService.showHint(
+              $scope.currentInteractionHints, isSupplementalCard);
+          };
+
+          $scope.isHintAvailable = function() {
+            var hintIsAvailable = (
+              HintManagerService.isCurrentHintUsable() &&
+              !HintManagerService.areAllHintsExhausted(
```

Comment:
I don't understand why the hint manager service can't just keep track of the hints. You could init()/reset() it with the list of hints when a new state is added on at the end, and then do stuff like getCurrentHint() etc.